autogluon/utils/tabular/ml/models/knn/knn_model.py

```diff
@@ -26,6 +27,7 @@ def preprocess(self, X):
         cat_columns = X.select_dtypes(['category']).columns
         X = X.drop(cat_columns, axis=1)  # TODO: Test if crash when all columns are categorical
         X = super().preprocess(X).fillna(0)
+        X = normalize(X, copy=True)
```

Comment:
Will do a review of the rest of the changes later, but just wanted to comment on this:

I believe this has to be done in a stateful manner or else the normalization done on train/val/test will differ.

I would also prefer if this change be its own separate PR as it is a major functionality change for KNN (It is not necessarily a strict improvement on all problems and it slows down KNN for both training and prediction, so it needs to be evaluated through a benchmark)